### PR TITLE
[CrowdSourcing] Turn annotation edits

### PR DESCRIPTION
**Pull request by jxmsML**
**Patch description**
A few issues for enabling open sourcing mephisto task dependent on turn_annotation_static task:
- unable to parse any quoted string (e,g,<span style="color: XXX") in task_description or task_title as it's re-processed in the jsx file since """ from Python is carrying over from ARG_STRING during pre-hydra era.
- Allow checkbox input type to be configurable, like the following (use radio button instead multi-label checkboxes)
Screen Shot 2020-10-28 at 2 34 55 PM
- Disable submit-button until the user input is validated. Before the change, the submit-button is grayish and turkers can still click to proceed to the next subtask even if the user input is NOT valid.
- Display correct closing left pane messages (before the change, it would show "You are currrently at conversation 4 / 3 " at the left pane after they submit the answers to the last subtask and before they return the HIT. Instead, the desired message would be:
Screen Shot 2020-10-28 at 2 20 54 PM

This pr also add a safety mephisto task that is dependent on the turn_annotation_static task.

**Testing steps**
CI

**Logs**

``` python projects/safety_recipes/human_safety_evaluation/run.py ```

Screen Shot 2020-10-29 at 3 40 16 PM
Screen Shot 2020-10-29 at 3 40 30 PM

**Other information**

### CODE REVIEW

File: projects/safety_recipes/human_safety_evaluation/run.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    STATIC_BLUEPRINT_TYPE,
+)
+from parlai.crowdsourcing.tasks.turn_annotations_static.util import run_static_task
+from parlai.crowdsourcing.utils.mturk import MTurkRunScriptConfig
+
+TASK_DIRECTORY = os.path.dirname(os.path.abspath(__file__))
+PARLAI_HOME = os.path.dirname(
+    os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+)
+BASE_TASK_DIRECTORY = os.path.join(
```
Comment (EricMichaelSmith) [inline projects/safety_recipes/human_safety_evaluation/run.py:25]: you can define this more easily (and in a way that is less likely to break in the future if the paths change) by importing the `run.py` file from `turn_annotations_static` and then setting `BASE_TASK_DIRECTORY = os.path.dirname(os.path.abspath(run.__file__))`

**Review by EricMichaelSmith — COMMENTED** [review]
Thanks Jing - this is great to have! I had a few tiny cleanup comments first but I think it's almost ready to go

File: projects/safety_recipes/human_safety_evaluation/run.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    PARLAI_HOME, 'parlai/crowdsourcing/tasks/turn_annotations_static'
+)
+
+defaults = [
```
[the remark this answers is not in the document]
Comment (jxmsML) [reply projects/safety_recipes/human_safety_evaluation/run.py:29]: good pt!

File: parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx [hunk 3]
```diff
@@ -120,6 +120,7 @@ var handleSubtaskSubmit = function (subtaskIndex, numSubtasks, initialTaskData,
     mephistoSubmit(window.workerAnswers);
   }
   showDisableCssNextButton();
+  setIndex(subtaskIndex + 1);
```
Comment (jxmsML) [inline parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx:123]: moving setIndex from old L233 to here (in old version, index (subtask index get bumped to next +1 regardless of whether the turker input is valid or not)

File: projects/safety_recipes/human_safety_evaluation/run.py (hunk 1 above)
Comment (jxmsML) [reply projects/safety_recipes/human_safety_evaluation/run.py]: good pt!

**Review by EricMichaelSmith — APPROVED** [review]
Looks good to me - thanks for those tweaks!

**Review by emilydinan — APPROVED** [review]
LGTM, thanks @jxmsML 🥳  

Can you update the projects page README with instructions for how to run this task with the safety stufF?

**Pull request merged**